railties/lib/rails/generators/rails/app/templates/config/puma.rb.tt

```diff
@@ -4,6 +4,15 @@
 # the maximum value specified for Puma. Default is set to 5 threads for minimum
 # and maximum; this matches the default thread size of Active Record.
 #
+
+if ENV['RAILS_ENV'] == 'development'
```

Comment:
While I can see value on this I don't think it should be the default of development. If you set like this by default you can end up with deadlocks in your application in case you have it making requests to the application and that is even more confusing to debug. I think we should accept a new ENV like DEBUG and set this when DEBUG is set.

Reply:
Great to know about the deadlocks. I've never seen this happen in the wild. Of course, the deadlock could be created by a different order of locking resources. 

1. What's another reason why having a single thread only will deadlock during development?
2. Maybe faster local testing time, non-debugging, is a reason to have more than one thread as the default?

I like a DEBUG env. How about

1. If the ENV 'RAILS_DEBUG' has a value of 'true', then use the debug settings.
2. Set this value to 'true' when running: `rails s --debug` or  `rails s -d` 

For reference, the `--help` for `rails s` would be, with this change (formatting off to highlight change):

```
➜  ~/tmp/rails7-test (main) rails --version
Rails 7.0.0
➜  ~/tmp/rails7-test (main) rails server --help
Usage:
  rails server -u [thin/puma/webrick] [options]

Options:
  -e, [--environment=ENVIRONMENT]              # Specifies the environment to run this server under (test/development/production).
  -d [--debug]                      # Start Rails with a single web thread for easier debugging
  -p, [--port=port]                            # Runs Rails on the specified port - defaults to 3000.
  -b, [--binding=IP]                           # Binds Rails to the specified IP - defaults to 'localhost' in development and '0.0.0.0' in other environments'.
  -c, [--config=file]                          # Uses a custom rackup configuration.
                                               # Default: config.ru
  -d, [--daemon], [--no-daemon]                # Runs server as a Daemon.
  -u, [--using=name]                           # Specifies the Rack server used to run the application (thin/puma/webrick).
  -P, [--pid=PID]                              # Specifies the PID file - defaults to tmp/pids/server.pid.
  -C, [--dev-caching], [--no-dev-caching]      # Specifies whether to perform caching in development.
      [--early-hints], [--no-early-hints]      # Enables HTTP/2 early hints.
      [--log-to-stdout], [--no-log-to-stdout]  # Whether to log to stdout. Enabled by default in development when not daemonized.
```

Reply:
@rafaelfranca if you approve this strategy, I'll update the PR.